android/expoview/src/main/java/host/exp/exponent/ExponentManifest.java

```diff
@@ -209,7 +209,7 @@ public void fetchManifest(final String manifestUrl, final ManifestListener liste
     String httpManifestUrl = uriBuilder.build().toString();
 
     // Fetch manifest
-    Request.Builder requestBuilder = ExponentUrls.addExponentHeadersToUrl(httpManifestUrl, manifestUrl.equals(Constants.INITIAL_URL), false);
+    Request.Builder requestBuilder = ExponentUrls.addExponentHeadersToUrl(httpManifestUrl, true, manifestUrl.equals(Constants.INITIAL_URL));
```

Comment:
These call sites are hard to read with the boolean args (when it's not a method like `setEnabled(true)` where it's clearer what true/false means). Similar to `addExponentHeadersToUrl(String)`, could you add `addExponentHeadersToManifestUrl(String, boolean isShellAppManifest)`? (That said, the second boolean arg isn't great either -- maybe we could remove it in a future PR? We've had requests to test out release channels in development too, so maybe we need to revisit how they work.)

Reply:
yep, I've refactored this, think it's better now.

I've left the `isShellAppManifest` arg there for now, since the client is this one weird special case where we don't want to send the release channel header with the request. It is possible to test release channels in the client if you pass the channel name as a query param in the URL. Is that what you meant by testing release channels in development?

Reply:
I think it's a variety of different requests -- wanting to test release channels with published projects in the Expo client, with published projects in ExpoKit, and also perhaps unpublished (in-development) projects in those environments as well, if you were to write some application logic that depends on the release channel.